src/Routing.jl

```diff
@@ -1,66 +1,76 @@
 # general methods for routing rivers
-export compute_streamflow!,
-    update_state_from_hillslope!, update_state_from_channel!
+export compute_streamflow,
+    compute_river_state, compute_channel_state, compute_hillslope_state
 
 using CSV, DataFrames, Dates, DSP, SpecialFunctions
 
 # methods
-function compute_streamflow!(
-    river_state::RS,
+function compute_streamflow(
+    initial_date_window::DateWindow,
     river_model::HCM,
     env::E,
-    start_date::Date,
     end_date::Date,
-) where {RS <: RiverState, HCM <: HillslopeChannelRiverModel, E <: Environment}
-    update_state!(river_state, river_model, env, start_date, end_date)
-    return compute_streamflow(river_state, env)
+) where {HCM <: HillslopeChannelRiverModel, E <: Environment}
+
+    date_window = initial_date_window
+    river_states = []
+    streamflows = []
+    while date_window.end_date <= end_date
+        # @info "computing streamflow over window [$(date_window.start_date),$(date_window.end_date)]"
+        river_state = compute_river_state(date_window, river_model, env)
+        push!(river_states, river_state)
+        push!(streamflows, compute_streamflow(river_state, env))
+        date_window = iterate(date_window)
```

Comment:
Note to self: this is a sliding window of length "history", in the examples, set to 50 days